roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -81,6 +90,19 @@ def run(self, tmp, task_vars):
             config_bytes = max(config.get(name, 0) for name in group_names) * 10**9
             recommended_bytes = config_bytes or recommended_bytes
 
+            # if an "upgrade" context is set, update the minimum disk requirement
+            # as this signifies an in-place upgrade - the node might have the
+            # required total disk space, but some of that space may already be
+            # in use by the existing OpenShift deployment.
+            context = get_var(task_vars, "r_openshift_health_checker_playbook_context", default="")
+            if context == "upgrade":
+                recommended_bytes = self.recommended_disk_upgrade_bytes.get(path, {})
+                if recommended_bytes:
+                    recommended_bytes = max(recommended_bytes.get(name, 0) for name in group_names)
+                else:
+                    # if not able to resolve a recommended value, default to non-upgrade recommendation
+                    recommended_bytes = max(recommendation.get(name, 0) for name in group_names)
```

Comment:
I think there are some subtle problems here:

- If the user configured the values, this block is overriding it;
- This line "resets" the value of `recommended_bytes` to "almost" what it was in line https://github.com/openshift/openshift-ansible/pull/4372/files#diff-dd45df49348ab2716e213a0ecc5b0f4dR91, though this time we ignore the user-config. That's because when we see the "upgrade" context we're hard-resetting the variable.